linter_driver_rustc/src/ast/rustc.rs

```diff
@@ -12,11 +11,8 @@ use linter_api::{
 
 use super::{ty::RustcTy, RustcLifetime, RustcSpan};
 
-#[expect(unused)]
 pub struct RustcContext<'ast, 'tcx> {
     pub(crate) lint_ctx: &'ast LateContext<'tcx>,
-    pub(crate) tcx: TyCtxt<'tcx>,
-    pub(crate) lint_store: &'tcx LintStore,
```

Comment:
I made this a single commit because I'm not sure it's better to remove the fields. `LateContext` has the `LintStore` and `TyCtxt` so it seems a bit redundant? We need the `LateContext` if we want to use the warn/error reporting of rustc.

Reply:
I remember not using `LateContext` due to some lifetime problems, as it doesn't live for `'tcx` but this might no longer be important. Even without it we could span a lint with [`struct_span_lint_hir`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.struct_span_lint_hir). That is also the function used by `LateContext` internally. But if everything works like this, it should be fine :)